includes/Gateway/Gift_Card.php

```diff
@@ -451,19 +451,24 @@ public function add_gift_card_fragments( $fragments ) {
 				} else {
 					$response['is_error'] = true;
 
-					if ( is_array( $gift_card_data ) ) {
+					if ( is_array( $gift_card_data ) && $this->debug_log() ) {
 						$log = new \WC_Logger();
 
 						foreach ( $gift_card_data as $square_error ) {
 							if ( $square_error instanceof \Square\Models\Error ) {
 								/** @var \Square\Models\Error $square_error */
-								$log->log( 'error', $square_error->getDetail() );
+								$log->log( 'error', 'Gift card error: ' . $square_error->getDetail(), array( 'source' => 'woocommerce-square-gift-card' ) );
 							}
 						}
 					}
 				}
 			} else {
 				$response['is_error'] = true;
+
+				if ( $this->debug_log() ) {
+					$log = new \WC_Logger();
+					$log->log( 'error', 'Gift card error: Payment token (woocommerce_square_gift_card_payment_token) is missing from session.', array( 'source' => 'woocommerce-square-gift-card' ) );
```

Comment:
Better to use `$this->get_plugin()->log()` or `wc_square()->log()` here to log this in specific file. Like we are doing at other places. Apply same to earlier log as well.

Reply:
Sure, this is now done. Thanks for the suggestion!